server/src/main/java/org/elasticsearch/action/bulk/WriteMemoryLimits.java

```diff
@@ -20,26 +20,68 @@
 package org.elasticsearch.action.bulk;
 
 import org.elasticsearch.common.lease.Releasable;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.util.concurrent.EsRejectedExecutionException;
 
 import java.util.concurrent.atomic.AtomicLong;
 
 public class WriteMemoryLimits {
 
+    // TODO: Adjust
+    public static final Setting<ByteSizeValue> MAX_INDEXING_BYTES =
+        Setting.memorySizeSetting("indices.write.limit", "10%", Setting.Property.NodeScope, Setting.Property.Dynamic);
```

Comment:
As discussed, let's make this a non-dynamic node setting. We also need to discuss whether we should white-list this setting in Cloud (so that users can change it there).

I think that we should not pick a setting name that is under the  "indices" namespace, but perhaps introduce something completely new, for example `indexing_limits.memory.limit`

Reply:
I changed the name and made it non-dynamic.